nomenclature/definition.py

```diff
@@ -136,22 +146,67 @@ def check_aggregate(self, df: IamDataFrame, **kwargs) -> None:
             error = pd.concat(lst)
             return error if not error.empty else None
 
-    def to_excel(
-        self, excel_writer, sheet_name=None, sort_by_code: bool = False, **kwargs
-    ):
-        """Write the *variable* codelist to an Excel sheet
+    def to_excel(self, excel_writer, sort_by_code: bool = False, **kwargs):
+        """Write the codelists to an xlsx spreadsheet
 
         Parameters
         ----------
         excel_writer : path-like, file-like, or ExcelWriter object
             File path as string or :class:`pathlib.Path`,
             or existing :class:`pandas.ExcelWriter`.
-        sheet_name : str, optional
-            Name of sheet that will have the codelist. If *None*, use the codelist name.
         sort_by_code : bool, optional
             Sort the codelist before exporting to file.
         **kwargs
             Passed to :class:`pandas.ExcelWriter` (if *excel_writer* is path-like).
         """
-        # TODO write all dimensions to the file
-        self.variable.to_excel(excel_writer, sheet_name, sort_by_code, **kwargs)
+        with pd.ExcelWriter(excel_writer, engine="xlsxwriter", **kwargs) as writer:
+
+            # create dataframe with attributes of the DataStructureDefinition
+            arg_dict = {
+                "project": self.working_dir.absolute().parts[-1],
+                "file created": time_format(datetime.now()),
+                "": "",
+            }
+            if self.repo is not None:
+                arg_dict.update(git_attributes("", self.repo))

```

Comment:
You could also include a check here that there are no unstaged chages in the git repository and only allow exporting of the Excel file if there's none.